test/ptp/ptp.go

```diff
@@ -304,6 +304,44 @@ var _ = Describe("[ptp]", func() {
 			waitUntilLogIsDetected(testPtpPod, 3*time.Minute, "Profile Name: slave")
 		})
 	})
+
+	Describe("PTP Metric", func() {
+		ptpRunningPods := []v1core.Pod{}
+		BeforeEach(func() {
+			ptpPods, err := client.Client.Pods(PtpLinuxDaemonNamespace).List(metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
```

Comment:
I would get rid of this and put the metric test under PTP e2e tests.
If we don't do that, we'll have to replicate all the logic there to check / configure ptp / discovery mode.
Here we are checking that the daemons are running, but we may even need to configure those before. The test will fail if we focus only on PTP metric because the setup won't happen.
Also, we rely on the test order execution assuming the setup was done before.